staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -1529,3 +1578,62 @@ func (g *fixtureTestGen) Init(c *generator.Context, w io.Writer) error {
 	}
 	return nil
 }
+
+// extractInnerValidations extracts all +k8s:inner validations for the given
+// subfield that were defined on the parent struct.The syntax for the tag is
+// +k8s:inner(subfield-name)=<validator-tag>=<args>
+func (td *typeDiscoverer) extractInnerValidations(subfield *types.Member, comments []string) (validators.Validations, error) {
+	var result validators.Validations
+
+	// Currently the format for +k8s:inner tag is:
+	// +k8s:inner(subfield-name)=<validator-tag>=<args> validator tag args..."
+	fieldTag := fmt.Sprintf("%s(%s)", innerTag, subfield.Name)
+	if tagVals, found := gengo.ExtractCommentTags("+", comments)[fieldTag]; found {
+		for _, tagVal := range tagVals {
+			// Extract any embedded validation rules.
+			fakeComments := []string{tagVal}
+			if innerValidations, err := td.validator.ExtractValidations(subfield.Type, fakeComments); err != nil {
+				return result, err
+			} else {
+				if !innerValidations.Empty() {
+					klog.V(5).InfoS("  found inner-validations", "field", subfield.Name, "n", innerValidations.Len())
+					result.Add(innerValidations)
+				}
+			}
+		}
+	}
+	return result, nil
+}
+
+// forEachField iterates over each exported field of the given type,
+// and performs the given operation.
+func forEachField(t *types.Type, op func(member types.Member, name, jsonName string) error) error {
+	for _, memb := range t.Members {
+		name := memb.Name
+		if len(name) == 0 { // embedded fields
+			if memb.Type.Kind == types.Pointer {
+				name = memb.Type.Elem.Name.Name
+			} else {
+				name = memb.Type.Name.Name
+			}
+		}
+		// Only do exported fields.
+		if unicode.IsLower([]rune(name)[0]) {
+			continue
+		}
+
+		// If we try to emit code for this field and find no JSON name, we
+		// will abort.
+		jsonName := ""
+		if commentTags, ok := tags.LookupJSON(memb); ok {
+			jsonName = commentTags.Name
+		}
+
+		klog.V(5).InfoS("field", "name", name, "jsonName", jsonName, "type", memb.Type)
```

Comment:
Super nit, but I would not log here, let the called function do that.

Reply:
done, removed this here and added it back into `doOneField` and `doOneChildField`